F proxy ytdlp

mfarre opened the pull request:
* [ x] I have read CONTRIBUTE.md

With this PR it is possible to configure yt_dlp with a proxy and with a flag check the proxy certificate or ignore it.

In the YAML configuration:
reading:
    yt_args:
        proxy: "<proxy_address>"
        proxy-check-certificate: False

mfarre committed:
do not crash when torchdistributed is not being used

mfarre committed:
allowing yt_dlp to run through a proxy

mfarre committed:
narrowing exception

rom1504 commented:
Hey, I think these points would help
* Making the ci pass
* Adding a test using a proxy (either an open one publicly available or starting a local one in the CI for testing purposes)
* Adding a note in the readme about proxy support
* Adding recommendation on what proxy / proxy software to use and speeds observed if using a proxy 

The first point is necessary. The rest would be nice

mfarre committed:
adding documentation + catching exception independently of torch version

mfarre commented:
thanks @rom1504, I think I addressed some changes that should make the CI go, added some comments in the readme and more general details in the yt-dlp specific examples section.

mfarre committed:
reformats

mfarre committed:
black/lint change
